File: ui/src/types/slots.ts
export interface ScheduleSlot {
  airline: string
  route: string
  airportFrom: string
  airportTo: string
  dayOfWeek: number
  dayLabel: string
  slot: string
  time: number
  length: number
  /** Vuelos observados en el periodo para esa franja. */
  flights: number
  observedRate: number
  /** Riesgo estimado por el modelo, que es lo que ordena el ranking. */
  risk: number
  band: string
}

export interface SlotsSummary {
  delayRate: number | null
  flightsInSelection: number
  totalFlights: number
  scheduleSlots: number
  slotsOverThreshold: number
  slotsInSelection: number
  rocAuc: number | null
  threshold: number
}

export interface SlotFilters {
  airline?: string
  route?: string
  dayOfWeek?: number
  slot?: string
}

File: api/app/schemas/slots.py
from typing import Optional

from app.schemas.base import CamelModel


class ScheduleSlot(CamelModel):
    airline: str
    route: str
    airport_from: str
    airport_to: str
    day_of_week: int
    day_label: str
    slot: str
    time: int
    length: int
    flights: int
    observed_rate: float
    risk: float
    band: str


class SlotsSummary(CamelModel):
    delay_rate: Optional[float]
    flights_in_selection: int
    total_flights: int
    schedule_slots: int
    slots_over_threshold: int
    slots_in_selection: int
    roc_auc: Optional[float]
    threshold: float
